src/Engine/ProtoCore/RuntimeData.cs

```diff
@@ -46,13 +46,18 @@ public class RuntimeData
         /// Map from a callsite's guid to a graph UI node. 		
         /// </summary>
         public Dictionary<Guid, Guid> CallSiteToNodeMap { get; }
- 		 
- #endregion
+        /// <summary>		
+        /// Map from a graph UI node to callsite identifiers. 		
+        /// </summary>
+        public Dictionary<Guid, List<string>> NodeToCallsiteIdentifiersMap { get; }
```

Comment:
Seems `internal` now, can we expose this later?

Reply:
Access modifier can be set more conservative but I would argue the other runtime dictionaries should be as well.

Reply:
I purposefully did not use `Dictionary<Guid, List<Guid>>` as the primary lookup key for the `CallsiteCache` dictionary was via the Callsite Identifier string.  Using the `Callsite.CallsiteID` Guid would require another series of ForEach loops or the creation of another map to the `Callsite` objects which did not seem advisable.   `CallsiteCache` is the critical map here and is used in many places.  The `NodeToCallsite` is only used in the `ReconcileTraceDataAndNotify()` which we should also look at core it's inefficiency.